tck/src/main/java/ee/jakarta/tck/data/framework/signature/DataSignatureTestRunner.java

```diff
@@ -0,0 +1,387 @@
+    // environment, you may use the property sigTestClasspath to specify
+    // where the API under test lives. This should almost never be used.
+    // Normally the API under test should be specified in the classpath
+    // of the VM running the signature tests. Use either the first
+    // comment or the one below it depending on which properties your
+    // signature tests need. Please do not use both comments.
+
+    public void signatureTest() throws Fault {
+        log.info("ConcurrencySigTest.signatureTest() called");
+        SigTestResult results = null;
+        String mapFile = null;
+        String packageListFile = null;
+        String signatureRepositoryDir = null;
+//        Properties mapFileAsProps = null;
+        try {
+            InputStream inStreamMapfile = DataSignatureTestRunner.class.getClassLoader()
+                    .getResourceAsStream(SIG_RESOURCE_PACKAGE.replace(".", "/") + "/" + SIG_MAP_NAME);
+            File mFile = writeStreamToTempFile(inStreamMapfile, "sig-test", ".map");
+            mapFile = mFile.getCanonicalPath();
+            log.info("mapFile location is :" + mapFile);
+
+            InputStream inStreamPackageFile = DataSignatureTestRunner.class.getClassLoader()
+                    .getResourceAsStream(SIG_RESOURCE_PACKAGE.replace(".", "/") + "/" + SIG_PKG_NAME);
+            File pFile = writeStreamToTempFile(inStreamPackageFile, "sig-test-pkg-list", ".txt");
+            packageListFile = pFile.getCanonicalPath();
+            log.info("packageFile location is :" + packageListFile);
+
+//            mapFileAsProps = getSigTestDriver().loadMapFile(mapFile);
+
+            InputStream inStreamSigFile = DataSignatureTestRunner.class.getClassLoader()
+                    .getResourceAsStream(SIG_RESOURCE_PACKAGE.replace(".", "/") + "/" + SIG_FILE_NAME);
+            File sigFile = writeStreamToSigFile(inStreamSigFile);
+            log.info("signature File location is :" + sigFile.getCanonicalPath());
+            signatureRepositoryDir = System.getProperty("java.io.tmpdir");
+
+        } catch (IOException ex) {
+            log.info("Exception while creating temp files :" + ex);
+        }
+
+        String[] packagesUnderTest = getPackages(testInfo.getVehicle());
+        String[] classesUnderTest = getClasses(testInfo.getVehicle());
+        String optionalPkgToIgnore = testInfo.getOptionalTechPackagesToIgnore();
+
+        // unlisted optional packages are technology packages for those optional
+        // technologies (e.g. jsr-88) that might not have been specified by the
+        // user.
+        // We want to ensure there are no full or partial implementations of an
+        // optional technology which were not declared
+        ArrayList<String> unlistedTechnologyPkgs = getUnlistedOptionalPackages();
+
+        // If testing with Java 9+, extract the JDK's modules so they can be used
+        // on the testcase's classpath.
```

Comment:
Jakarta Data will not ever be tested with Java 8 and earlier if that helps to optimize the subsequent code